oc-chef-pedant/spec/api/cookbooks/read_spec.rb

```diff
@@ -325,7 +325,10 @@ def self.expects_response_of_400_with(message, _value)
               http.ca_file     = Pedant::Config.ssl_ca_file     if Pedant::Config.ssl_ca_file
             end
 
-            response = http.get(uri.request_uri, {})
+            # NOTE: Chef::ServerAPI always sets the Host
+            # header to HOSTNAME:PORT. We do the same here to avoid sigv4
+            # signing issues.
+            response = http.get(uri.request_uri, {"Host" => "#{uri.hostname}:#{uri.port}"})
```

Comment:
I know it is silly, but maybe we can pull this out to a utility function so that we can just have this comment in one place.  Also, the comment could do more to explain what the signing issue is.  For instance:

`Chef::ServerAPI always sets the Host header to HOSTNAME:PORT. We do the same thing here since the Version 4 AWS Signature scheme uses the Host header as part of the canonically signed content.`

Also, am I correct that this isn't strictly necessary since we are also doing the Host-header fallback handling in bookshelf for compatibility?

Reply:
It is strictly necessary in the case of using S3 vs bookshelf.

Reply:
Will investigate pulling this out to a utility function.

Reply:
Factored-out things into a utility function.  Let me know if it's what you had in mind.